src/core/examples.js

```diff
@@ -6,69 +6,71 @@
 // be used by a containing shell to extract all examples.
 
 import { pub } from "core/pubsubhub";
+import "deps/hyperhtml";
 import css from "deps/text!core/css/examples.css";
 
 export const name = "core/examples";
+const isEmpty = str => /^\s*$/.test(str);
 
-function makeTitle(conf, $el, num, report) {
-  const txt = num > 0 ? " " + num : "";
-  const $tit = $(
-    `<div class='example-title'><span>${conf.l10n.example}${txt}</span></div>`
-  );
-  report.title = $el.attr("title");
-  if (report.title) {
-    $tit.append(
-      $("<span style='text-transform: none'>: " + report.title + "</span>")
-    );
-    $el.removeAttr("title");
-  }
-  $tit.addClass("marker");
-  return $tit;
+function makeTitle(conf, elem, num, report) {
+  report.title = elem.title;
+  if (report.title) elem.removeAttribute("title");
+  const number = num > 0 ? " " + num : "";
+
+  return hyperHTML`
+  <div class='marker example-title'><span>${conf.l10n.example}${number}</span>${
+    report.title ? hyperHTML`<span>: ${report.title}</span>` : ""
+  }</div>`;
 }
 
-export function run(conf, doc, cb) {
-  const $exes = $("pre.example, pre.illegal-example, aside.example");
-  let num = 0;
-  if ($exes.length) {
-    $(doc)
-      .find("head link")
-      .first()
-      .before($("<style/>").text(css));
-    $exes.each((i, ex) => {
-      const $ex = $(ex);
-      const report = { number: num, illegal: $ex.hasClass("illegal-example") };
-      if ($ex.is("aside")) {
-        num++;
-        const $tit = makeTitle(conf, $ex, num, report);
-        $ex.prepend($tit);
-        pub("example", report);
-      } else {
-        const inAside = !!$ex.parents("aside").length;
-        if (!inAside) num++;
-        // reindent
-        const lines = $ex.html().split("\n");
-        while (lines.length && /^\s*$/.test(lines[0])) lines.shift();
-        while (lines.length && /^\s*$/.test(lines[lines.length - 1]))
-          lines.pop();
-        const matches = /^(\s+)/.exec(lines[0]);
-        if (matches) {
-          const rep = new RegExp("^" + matches[1]);
-          for (let j = 0; j < lines.length; j++) {
-            lines[j] = lines[j].replace(rep, "");
-          }
+export function run(conf) {
+  const examples = document.querySelectorAll(
+    "pre.example, pre.illegal-example, aside.example"
+  );
+  if (!examples.length) return;
+
+  document.head.insertBefore(
+    hyperHTML`<style>${css}</style>`,
+    document.querySelector("link")
+  );
+
+  let number = 0;
+  examples.forEach(example => {
+    const illegal = example.classList.contains("illegal-example");
+    const report = { number, illegal };
+    if (example.localName === "aside") {
+      ++number;
+      const title = makeTitle(conf, example, number, report);
+      example.insertBefore(title, example.firstChild);
+      pub("example", report);
+    } else {
+      const inAside = !!example.closest("aside");
+      if (!inAside) ++number;
+
+      // reindent
```

Comment:
So this reindentation algorithm is duplicated across files including `webidl.js`. I think I have seen another one elsewhere but I forgot where it was.

Reply:
Oh yes. We should move it to core/utils then, right? BTW, I found it only in webidl and here (no other place).

Also, I think the re-indent algorithm is slightly different (`leastIndent` in webidl vs indentation of `lines[0]` in examples). Which one should we use?

Reply:
I personally prefer `leastIndent` but both probably will work.